tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -0,0 +1,293 @@
+/// \file ROOT/RNTupleProcessor.hxx
+/// \ingroup NTuple ROOT7
+/// \author Florine de Geus <florine.de.geus@cern.ch>
+/// \date 2024-03-26
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2024, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RNTupleProcessor
+#define ROOT7_RNTupleProcessor
+
+#include <ROOT/REntry.hxx>
+#include <ROOT/RError.hxx>
+#include <ROOT/RNTupleDescriptor.hxx>
+#include <ROOT/RNTupleModel.hxx>
+#include <ROOT/RNTupleUtil.hxx>
+#include <ROOT/RPageStorage.hxx>
+#include <ROOT/RSpan.hxx>
+
+#include <memory>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace ROOT {
+namespace Experimental {
+
+/// Helper type representing the name and storage location of an RNTuple.
+struct RNTupleSourceSpec {
+   std::string fName;
+   std::string fStorage;
```

Comment:
You mean add the locator to this struct? We could do that, although at this point I'm not sure what that would add since it's already handled by the page source.